### ISSUE TITLE
Add keyboard navigation to this sim

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/gravity-force-lab-basics#15]
@emily-phet suggested that it would be great to include keyboard navigation in this sim as part of its initial deployment.

**Comment by jessegreenberg** [comment]
Here is the latest version with keyboard navigation:
http://www.colorado.edu/physics/phet/dev/html/gravity-force-lab-basics/1.0.0-dev.4/gravity-force-lab-basics_en.html?accessibility

The items missing keyboard nav are the masses and the spin boxes.

**Comment by emily-phet** [comment]
@jessegreenberg, great! Do you have an idea of how the masses and spin boxes may be implemented? Is the approach we had discussed before (and I think you started the implementation on) still seem feasible, or are there open questions/issues?

**Comment by jessegreenberg** [comment]
Implementation does seem relatively straight forward, but there are a few questions:

- What should the focus highlights look like for the spin boxes and masses?  Should focus highlights on spin boxes include labels? Is a box around the puller and sphere OK for the mass? Or do we want a unique shape that contours around the sphere and the puller robot?

- Typical HTML spin boxes allow user to enter values with number keys.  Can we disable that here for simplicity or do we want to include entering numbers with the keyboard?

- The implementation we tried for the masses didn't work in anything other than chrome, we will have to investigate why.

**Comment by jessegreenberg** [comment]
> Typical HTML spin boxes allow user to enter values with number keys. Can we disable that here

To clarify, my reasoning for this is that our number pickers don't have a cursor (HTML number inputs do), so entering values won't be the expected interaction.

**Comment by jessegreenberg** [comment]
I have been assuming that we would use the arrow keys to change NumberSpinner values.  But maybe we want to to tab to the up/down arrows and activate with enter/spacebar?

**Comment by jessegreenberg** [comment]
@emily-phet here is an initial version of the sim with keyboard navigation for the number spinners and the masses.  I tested on Chrome, Firefox, and Safari it seems to be working on those platforms.

http://www.colorado.edu/physics/phet/dev/html/gravity-force-lab-basics/1.0.0-dev.6/gravity-force-lab-basics_en.html?accessibility

Any thoughts on this version or on the comments in https://github.com/phetsims/gravity-force-lab-basics/issues/15?_pjax=%23js-repo-pjax-container#issuecomment-312137038?

**Comment by jessegreenberg** [comment]
The navigation order is probably very wrong, could definitely be changed, but the masses should be working with the keyboard.

**Comment by emily-phet** [comment]
@jessegreenberg 
- Order: Mass1, Mass2, mass1 spin box, mass2 spin box, control panel, reset all
- Focus highlight: 
      Masses - I like the focus as you have it now
      Spin boxes - make focus rectangle go around spin box only (just outside the little arrows on the top/bottom) and not including the units. Would be nice if the focus rectangle had rounded corners, to mimic the shape of the spin boxes.

**Comment by emily-phet** [comment]
@jessegreenberg - sorry, I missed some of your questions yesterday.
- I think it's fine to disable entering numbers for the spin boxes
- Moving the masses seems to work fine for me in Safari, can you describe what isn't working outside of Chrome?

**Comment by jessegreenberg** [comment]
Thanks @emily-phet, assigning back to me.

**Comment by jessegreenberg** [comment]
@emily-phet I think I have addressed these things, here is a new version, would you mind reviewing?

http://www.colorado.edu/physics/phet/dev/html/gravity-force-lab-basics/1.0.0-dev.7/gravity-force-lab-basics_en.html?accessibility

Regarding:
> Moving the masses seems to work fine for me in Safari, can you describe what isn't working outside of Chrome?

This time things seem to be working in all browsers!  When we first explored this in February, it wasn't working as well.

**Comment by emily-phet** [comment]
@jessegreenberg - Looks great to me!

**Comment by jessegreenberg** [comment]
Great, sounds like we are done here then!  Closing this issue, others will be created if there are requests for additional improvements.
